services/cert/provider.go

```diff
@@ -0,0 +1,168 @@
+	apiResponse.LogInfo()
+	return provider, apiResponse, err
+}
+
+//nolint:golint
+func (c *Client) ListProviders(ctx context.Context, location string) (certmanager.ProviderReadList, *certmanager.APIResponse, error) {
+	c.modifyConfigURL(location)
+	providers, apiResponse, err := c.sdkClient.ProviderApi.ProvidersGet(ctx).Execute()
+	apiResponse.LogInfo()
+	return providers, apiResponse, err
+}
+
+//nolint:golint
+func (c *Client) CreateProvider(ctx context.Context, providerPostData certmanager.ProviderCreate, location string) (certmanager.ProviderRead, *certmanager.APIResponse, error) {
+	c.modifyConfigURL(location)
+	provider, apiResponse, err := c.sdkClient.ProviderApi.ProvidersPost(ctx).ProviderCreate(providerPostData).Execute()
+	apiResponse.LogInfo()
+	return provider, apiResponse, err
+}
+
+//nolint:golint
+func (c *Client) UpdateProvider(ctx context.Context, providerID, location string, providerPatchData certmanager.ProviderPatch) (certmanager.ProviderRead, *certmanager.APIResponse, error) {
+	c.modifyConfigURL(location)
+	provider, apiResponse, err := c.sdkClient.ProviderApi.ProvidersPatch(ctx, providerID).ProviderPatch(providerPatchData).Execute()
+	apiResponse.LogInfo()
+	return provider, apiResponse, err
+}
+
+//nolint:golint
+func (c *Client) DeleteProvider(ctx context.Context, providerID, location string) (*certmanager.APIResponse, error) {
+	c.modifyConfigURL(location)
+	apiResponse, err := c.sdkClient.ProviderApi.ProvidersDelete(ctx, providerID).Execute()
+	apiResponse.LogInfo()
+	return apiResponse, err
+}
+
+//nolint:golint
+func (c *Client) IsProviderReady(ctx context.Context, d *schema.ResourceData) (bool, error) {
+	providerID := d.Id()
+	location := d.Get("location").(string)
+	provider, _, err := c.GetProvider(ctx, providerID, location)
+	if err != nil {
+		return false, fmt.Errorf("error checking certificate manager provider status: %w", err)
+	}
+	if provider.Metadata == nil || provider.Metadata.State == nil {
+		return false, fmt.Errorf("metadata or state is empty for certificate manager provider with ID: %v", d.Id())
+	}
+	if utils.IsStateFailed(*provider.Metadata.State) {
+		return false, fmt.Errorf("error while checking if auto-certificate provider is ready, provider ID: %v, state: %v", *provider.Id, *provider.Metadata.State)
+	}
+	return strings.EqualFold(*provider.Metadata.State, constant.Available), nil
+}
+
+//nolint:golint
+func (c *Client) IsProviderDeleted(ctx context.Context, d *schema.ResourceData) (bool, error) {
+	providerID := d.Id()
+	location := d.Get("location").(string)
+	provider, apiResponse, err := c.GetProvider(ctx, providerID, location)
+	if err != nil {
+		if apiResponse.HttpNotFound() {
+			return true, nil
+		}
+		return false, fmt.Errorf("error while checking deletion status for certificate manager provider with ID: %v, error: %w", d.Id(), err)
+	}
+	if provider.Metadata != nil && provider.Metadata.State != nil {
+		if utils.IsStateFailed(*provider.Metadata.State) {
```

Comment:
can merge these into a single `if`

Reply:
@cbeti-ionos done in ec747230649ad96067a92e1d3b11d469d3eda2f1